specs/Print.md

```diff
@@ -0,0 +1,473 @@
+
+        await webView.CoreWebView2.PrintWithSettingsAsync(printSettings);
+            MessageBox.Show(this, "Printing is succeeded", "Print with Settings");
+    }
+}
+```
+
+## PrintToPdfStream
+
+You can use `PrintToPdfStream` method to display as a preview in a custom print dialog. The app can also use this API to configure the custom print preview dialog unlike confining print preview dialog to the webview2 control.
+
+// This example prints the Pdf data of the current page to a stream.
+
+```cpp
+// Function to display current page pdf data in a custom print preview dialog.
+static void DisplayPdfDataInPrintDialog(IStream* pdfData)
+{
+    // You can display the printable pdf data in a custom print preview dialog to the end user.
+}
+
+bool AppWindow::PrintToPdfStream()
+{
+    wil::com_ptr<ICoreWebView2_15> webView2_15;
+    CHECK_FAILURE(m_webView->QueryInterface(IID_PPV_ARGS(&webView2_15)));
+    CHECK_FEATURE_RETURN(webView2_15);
+
+    wil::com_ptr<ICoreWebView2Environment6> webviewEnvironment6;
+    CHECK_FAILURE(m_webViewEnvironment->QueryInterface(IID_PPV_ARGS(&webviewEnvironment6)));
+    CHECK_FEATURE_RETURN(webviewEnvironment6);
+
+    wil::com_ptr<ICoreWebView2PrintSettings> printSettings = nullptr;
+    CHECK_FAILURE(webviewEnvironment6->CreatePrintSettings(&printSettings));
+
+    wil::com_ptr<ICoreWebView2PrintSettings2> printSettings2;
+    CHECK_FAILURE(printSettings->QueryInterface(IID_PPV_ARGS(&printSettings2)));
+    CHECK_FEATURE_RETURN(printSettings2);
+
+    CHECK_FAILURE(printSettings2->put_Orientation(COREWEBVIEW2_PRINT_ORIENTATION_PORTRAIT));
+    CHECK_FAILURE(printSettings2->put_ShouldPrintBackgrounds(true));
+
+    CHECK_FAILURE(
+        webView2_15->PrintToPdfStream(
+            printSettings2.get(),
+            Callback<ICoreWebView2PrintToPdfStreamCompletedHandler>(
+                [this](HRESULT errorCode, IStream* pdfData) -> HRESULT
+                {
+                    CHECK_FAILURE(errorCode);
+                    DisplayPdfDataInPrintDialog(pdfData);
+                    return S_OK;
+                })
+                .Get()));
+     return true;
+}
+```
+
+```c#
+async void PrintToPdfStream()
+{
+    CoreWebView2PrintSettings printSettings = null;
+    printSettings = WebViewEnvironment.CreatePrintSettings();
+    printSettings.Orientation = CoreWebView2PrintOrientation.Portrait;
+    printSettings.ShouldPrintBackgrounds = true;
+
+    MemoryStream pdfStream = new MemoryStream();
+    System.IO.Stream stream = await webView.CoreWebView2.PrintToPdfStreamAsync(printSettings);
+    DisplayPdfDataInPrintDialog(pdfData);
+}
+
+// Function to display current page pdf data in a custom print preview dialog.
+void DisplayPdfDataInPrintDialog(Stream pdfData)
+{
+    // You can display the printable pdf data in a custom print preview dialog to the end user.
+}
+```
+
+# API Details
+
+## Win32 C++
+``` cpp
+/// Specifies the print dialog kind.
+[v1_enum]
+typedef enum COREWEBVIEW2_PRINT_DIALOG_KIND {
```

Comment:
is this API COM only, or is there a WinRT API that accompanies it?

Reply:
The WebView2 APIs are a bit different. We have a COM API (that's this IDL and the c++ sample code above), a .NET API, and a WinRT API (that's the MIDL3 at the bottom and the C# samples)